mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -111,6 +113,17 @@ public StoreRightManager getStoreRightManager() {
         public InMemoryPerUserMaxQuotaManager getMaxQuotaManager() {
             return maxQuotaManager;
         }
+
+        public MessageIdManager createMessageIdManager() {
```

Comment:
it looks like a default choice, there may be several MessageIdManager, how can you decide which is the right one?

Reply:
I rewrote this, now this method is turned into:

```
    public MessageIdManager getMessageIdManager() {
        return storeMessageIdManager;
    }
```

BTW right now there is only one implementation for MessageIdManager which IMO makes a possible choice useless.

When needed we can very well add a method in the factory for allowing customizing this.